crowbar_framework/app/controllers/backup_controller.rb

```diff
@@ -0,0 +1,172 @@
+        format.html { redirect_to backup_path }
+      else
+        msg = I18n.t(".invalid_filename", scope: "backup.index")
+        render json: { error: msg }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+      end
+    end
+  end
+
+  #
+  # Restore
+  #
+  # Provides the restful api call for
+  # /utils/backup/restore   POST   Trigger a restore
+  def restore
+    @backup = Backup.new(
+      name: params[:name],
+      created_at: params[:created_at]
+    )
+
+    respond_to do |format|
+      if @backup.valid?
+        ret = @backup.restore
+        if ret[:status] == :ok
+          format.json { head :ok }
+          format.html { redirect_to backup_path }
+        else
+          format.html do
+            flash[:alert] = ret[:msg]
+            redirect_to backup_path
+          end
+          format.json { render json: ret[:msg] }
+        end
+      else
+        msg = I18n.t(".invalid_backup", scope: "backup.index")
+        render json: { error: msg }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+      end
+    end
+  end
+
+  #
+  # Download
+  #
+  # Provides the restful api call for
+  # /utils/backup/download/:name/:created_at 	GET 	Download a backup
+  def download
+    @backup = Backup.new(
+      name: params[:name],
+      created_at: params[:created_at]
+    )
+    respond_to do |format|
+      if @backup.exist?
+        format.any do
+          send_file(
+            @backup.path,
+            filename: @backup.filename
+          )
+        end
+      else
+        msg = I18n.t(".missing_backup", scope: "backup.index")
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+        format.json { render json: msg }
+      end
+    end
+  end
+
+  #
+  # Upload
+  #
+  # Provides the restful api call for
+  # /utils/backup/upload   POST   Upload a backup
+  def upload
+    file = params[:upload]
+    filename, created_at = Backup.filename_time(file.original_filename)
+    @backup = Backup.new(
+      name: filename,
+      created_at: created_at
+    )
+
+    respond_to do |format|
+      if @backup.valid?
+        @backup.upload(file)
+
+        format.html { redirect_to backup_path }
+        format.json { head :ok }
+      else
+        format.html { flash[:alert] = @backup.errors }
+        format.json { render json: @backup.errors }
+      end
+    end
+  end
+
+  #
+  # Delete Backup
+  #
+  # Provides the restful api call for
+  # data-confirm method delete
+  # /utils/backup/delete 	DELETE 	Delete a backup
+  def delete
+    @backup = Backup.new(
+      name: params[:name],
+      created_at: params[:created_at]
+    )
+
+    respond_to do |format|
+      if @backup.valid?
+        @backup.delete
+
+        format.json { head :ok }
+        format.html { redirect_to backup_path }
+      else
+        render json: { error: @backup.errors }
```

Comment:
It's wrong to remove `format.json {}`, I just wanted to have the `error` key, e.g. `format.json { render json: { error: msg } }`